content/blog/2020-01-ggplot2-3-3-0/index.Rmarkdown

```diff
@@ -0,0 +1,231 @@
+      colour = stage(start = drv, after_scale = alpha(colour, 0.5))
+    ), 
+    fill = NA, size = 4
+  )
+```
+
+`stage()` is definitely not something you need every day, but it does resolve a range of issues (like the one above) and it is nice to have full transparency and control over the aesthetic evaluation.
+
+### More extensible theming
+The possibility to extend coordinate systems and faceting has created a need to allow new theme elements to be specified (along with inheritance). One could envision e.g. a faceting setup that related panels with arrows, and in order to control the look of the arrows we would need to allow a call like `theme(panel.arrow = element_line(...))`. With this release we finally provide a way for extension developers to register new theme elements using `register_theme_elements()`. We urge extension developers to namespace the theme elements they create in order to avoid name clashes. If the above facet was part of the ggrelfacet package the registration would look something like this:
+
+```{r, eval=FALSE}
+register_theme_elements(
+  ggrelfacet.panel.arrow = element_line(
+    size = 3, arrow = arrow()
+  ),
+  element_tree = list(
+    ggrelfacet.panel.arrow = el_def("element_line", "line")
+  )
+)
+```
+
+The first argument defines the default look of the element, and the second one provides the type and inheritance (must be an `"element_line"` object, and will inherit from the `"line"` element). If possible you should try to inherit from some base element and only do general modifications, as that will mean that your new theme elements will fit into whatever theme your user is using. Being overly specific will result in elements that may fit into one theme but look out of place in all others
+
+The call above would be added to the package `.onLoad()` function to make sure the theme element was registered when the package was used, but you will of course also need to create extensions that actually use the theme element for it to have any effect.
+
+### Better contour calculations
+A recurring request has been to allow for filled contours. This has been problematic in the past because `stat_density_2d()` only calculated the outer contour and didn't cut out any inner holes or next level contours. This meant that if either the fill was partly transparent, or if the level contained any holes, the output would not be a true representation of the data. For an example of this, we can look at our favorite volcano:
+
+```{r}
+volcano_long <- data.frame(
+  x = as.vector(col(volcano)),
+  y = as.vector(row(volcano)),
+  z = as.vector(volcano)
+)
+
+ggplot(volcano_long, aes(x, y, z = z)) + 
+  geom_polygon(aes(fill = stat(level)), alpha = 0.5, stat = "contour") + 
+  guides(fill = "legend")
+```
+
+There is so much wrong going on here that we won't go through it. In any case this is all in the past because ggplot2 has moved on to using the new isoband package and now provides a geom for filled contours specifically:
```

Comment:
```suggestion
There is so much wrong going on here that we won't go through it. In any case, this is all in the past because ggplot2 has moved on to using the new isoband package and now provides a geom for filled contours specifically:
```